### ISSUE TITLE
PoC with crystal 0.24.2

### ISSUE THREAD

**Issue by gummybears** [ysbaddaden/crystal_library#1]
Tried your PoC with crystal 0.24.2 on linux 64 bits. Alas did not work, there is no compile option, only a build option. With the build option it does not work
```
user1 ~/download/try/crystal_library (master) $ make CRYSTAL_ROOT=/opt/crystal
CRYSTAL_PATH="src:/opt/crystal/share/crystal/src" /opt/crystal/bin/crystal build \
	--release --cross-compile -o libcounter src/libcounter.cr
cc 'libcounter.o' -o 'libcounter'  -rdynamic  -lpcre -lgc -lpthread /opt/crystal/share/crystal/src/ext/libcrystal.a -levent -lrt -ldl -L/usr/lib -L/usr/local/lib
ar rcs libcounter.a libcounter.o
user1 ~/download/try/crystal_library (master) $ make test
cc test/counter_test.c ./libcounter.a /opt/crystal/share/crystal/src/ext/libcrystal.a -lgc -ldl -levent -lpcre -o counter_test
./libcounter.a(libcounter.o): In function `main':
/opt/crystal/share/crystal/src/crystal/main.cr:134: multiple definition of `main'
/tmp/ccowU1tv.o:counter_test.c:(.text+0x0): first defined here
/usr/bin/ld: cannot find -lgc
collect2: error: ld returned 1 exit status
Makefile:18: recipe for target 'counter_test' failed
make: *** [counter_test] Error 1
```

**Comment by ysbaddaden** [comment]
Fixed in 919400383a60205635117f26aba82723b96ddaef. You'll may have to tweak a bunch of variables for `make` to pass. By default they expect latest official packages (binary in `/usr/lib/crystal/bin/crystal`, stdlib in `/usr/share/crystal/src`, and so on).

For example with old packages (or tarballs):

```console
$ make \
  CRYSTAL=/opt/crystal/embedded/bin/crystal \
  CRYSTAL_LIB=/opt/crystal/embedded/lib \
  CRYSTAL_SRC=/opt/crystal/src
```

**Comment by gummybears** [comment]
Tested new code, I am able to run the examples. Thanks
